sys/posix/sockets/posix_sockets.c

```diff
@@ -862,19 +877,34 @@ ssize_t recvfrom(int socket, void *restrict buffer, size_t length, int flags,
     return res;
 }
 
-ssize_t sendto(int socket, const void *buffer, size_t length, int flags,
-               const struct sockaddr *address, socklen_t address_len)
+ssize_t recvfrom(int socket, void *restrict buffer, size_t length, int flags,
+                 struct sockaddr *restrict address,
+                 socklen_t *restrict address_len)
 {
     socket_t *s;
+    int res;
+
+    mutex_lock(&_socket_pool_mutex);
+    s = _get_socket(socket);
+    mutex_unlock(&_socket_pool_mutex);
```

Comment:
Can we agree that this is a porting PR, not a mutex-fix PR? ;-) This can be done as a follow-up.